src/liquid/blocks/when.cr

```diff
@@ -2,7 +2,7 @@ require "./block"
 
 module Liquid::Block
   class When < InlineBlock
-    SIMPLE_EXP = /^\s*when ((?:\s*"([^"]+)",?)+)\s*$/
+    SIMPLE_EXP = /^\s*when ((?:\s*['"]([^"]+)['"],?)+)\s*$/
```

Comment:
This accepts things like `when 'potato"`.

Reply:
This and `when "potato'` are good negative test cases.

Reply:
it also do not accept things like `when 'pota"to'`